aries-site/src/examples/templates/dashboards/content/chartCards/utils/statistics.js

```diff
@@ -0,0 +1,108 @@
+/* Utility functions for statistics used in charting */
+
+export const mean = series =>
+  series.reduce((acc, cur) => acc + cur, 0) / series.length;
+
+export const linearRegression = pairs => {
+  /* Finds the linear regression formula using least-squares method
+   * https://en.wikipedia.org/wiki/Simple_linear_regression#Fitting_the_regression_line
+   * https://en.wikipedia.org/wiki/Ordinary_least_squares#Linear_model
+   */
+
+  if (typeof pairs[0].x !== 'number') {
+    console.error(
+      `${pairs[0].x} is not of type === 'number'. Linear regression
+requires categorical variables (e.g. dates) to be converted to numerical
+representations.`,
+    );
+    return undefined;
+  }
+
+  const n = pairs.length;
+  const independents = pairs.map(pair => {
+    if (typeof pair.x !== 'number') {
+      console.error(
+        `${pair.x} is not of type === 'number'. Linear regression
+requires categorical variables (e.g. dates) to be converted to numerical
+representations.`,
+      );
+    }
+    return pair.x;
+  });
+  const dependendents = pairs.map(pair => pair.y);
+  const sumX = independents.reduce((acc, cur) => acc + cur, 0);
+  const sumY = dependendents.reduce((acc, cur) => acc + cur, 0);
+  const sumXX = independents.reduce((acc, cur) => acc + cur * cur, 0);
+  const sumXY = pairs.reduce((acc, cur) => acc + cur.x * cur.y, 0);
+  const sumYY = dependendents.reduce((acc, cur) => acc + cur * cur, 0);
+  // console.log(sumX, sumY, sumXX, sumXY, sumYY);
```

Comment:
remove this?

Reply:
No! It's very important! 🤪 

Yes, I will remove.

Reply:
Done.